astropy/coordinates/baseframe.py

```diff
@@ -789,6 +790,11 @@ def _get_representation_info(cls):
         # This exists as a class method only to support handling frame inputs
         # without units, which are deprecated and will be removed.  This can be
         # moved into the representation_info property at that time.
+        try:
+            if cls._representation_info_cachetick == MetaBaseRepresentation.REPRESENTATION_CLASSES_CACHE_TICK:
+                return cls._repr_attrs
+        except:
+            pass
 
         repr_attrs = {}
```

Comment:
This first stanza would seem to have no business being on the class at all, as it only deals with representations. Might it make sense to move this to `representation.py`, perhaps adding a new dict with this information, and updating it on the metaclass?